components/feature/download/src/main/java/mozilla/components/feature/download/DownloadManagerFeature.kt

```diff
@@ -0,0 +1,61 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.download
+
+import android.Manifest
+import android.annotation.SuppressLint
+import android.support.annotation.RequiresPermission
+import mozilla.components.browser.session.Download
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+
+/**
+ * Feature implementation for proving download functionality for the selected session.
+ */
+class DownloadManagerFeature(
+    private val downloadManager: DownloadManager,
+    private val sessionManager: SessionManager
+) : SessionManager.Observer, Session.Observer {
```

Comment:
As there is only one `DownloadManager` how about adding that as default parameter:

```Kotlin
class DownloadManagerFeature(
    private val sessionManager: SessionManager,
    private val downloadManager: DownloadManager = DownloadManager()
) : SessionManager.Observer, Session.Observer {
```

If you see the need for multiple download manager implementations in the future (I guess that makes sense) then I'd expect `DownloadManager` to be an interface and your implementation to have a different name (`AndroidDownloadManager` - but then your `typealias` may break 😜).

Reply:
Let's keep it simple for now :), I will the default parameter 😄

Reply:
But I will need an extra parameter (context) for creating a` DownloadManager(private val applicationContext: Context)`

Reply:
@pocmo it wont be worst to have an extra parameter, just for creating a default value. 

```kotlin
class DownloadManagerFeature(
    context:Context,
    private val downloadManager: DownloadManager = DownloadManager(context),
    private val sessionManager: SessionManager
) : SessionManager.Observer, Session.Observer
```

👍 
``` kotlin
       val feature = DownloadManagerFeature(context, sessionManager = sessionManager)
```

👎  the parameter is never been used
```kotlin
        val manager = DownloadManager(context)
        val feature = DownloadManagerFeature(context, manager, sessionManager)
```

Should I just add a context to the constructor and internally create the download manager? 

```kotlin
class DownloadManagerFeature(
    context: Context,
    private val sessionManager: SessionManager
) : SessionManager.Observer, Session.Observer {
    private val downloadManager: DownloadManager = DownloadManager(context)
```

Reply:
Ah, I see. Mh. Yeah, I think creating the `DownloadManager` internally may be a good workaround - as long as this doesn't affect testability? :)

Once we have the need to inject other DownloadManager things we can open that up again.

Reply:
👍